src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+            return;
+        case COMMAND_EVENT_WORD:
+            executeAddEvent();
+            return;
+        case NO_COMMAND:
+            displayToUser(ERROR_NO_COMMAND_RECEIVED);
+            return;
+        default:
+            displayToUser(ERROR_INVALID_COMMAND_RECEIVED);
+            return;
+        }
+    }
+
+    /**
+     * Extracts the command word from user input. 
+     * 
+     * @return Command extracted. 
+     */
+    private static String getCommand() {
+        String commandWord = userCommand.split(" ")[0].toLowerCase();
+        if (userCommand.length() > 4) {
+            userCommand = userCommand.substring(userCommand.indexOf(" ") + 1, userCommand.length());
+        } else {
+            userCommand = null;
+        }
+        return commandWord;
+    }
+
+    /**
+     * Exits the program. 
+     */
+    private static void executeExitProgramRequest() {
+        System.exit(0);
+    }
+
+    /**
+     * Lists all tasks current in the tasks list. 
+     * Shows the type of task (T, D, E) and marks X if the task is done. 
+     */
+    private static void executeListAllTasks() {
+        displayToUser(tasks);
+    }
+
+    /**
+     * Checks if task number is valid. 
+     * 
+     * @param taskNumber Task number. 
+     * @return True if number is valid and false if invalid. 
+     */
+    private static boolean isTaskNumberValid(int taskNumber) {
+        return taskNumber >= 0 && taskNumber < sizeOfTaskList;
+    }
+
+    /**
+     * Marks a given task number as done. 
+     * 
+     * @param taskNumber The task number to mark as done (starting with 1). 
+     */
+    private static void executeMarkTask(int taskNumber) {
+        tasks[taskNumber].setTaskStatus();
+    }
+
+    /**
+     * Displays the success message after marking task as done. 
+     * 
+     * @param taskNumber The task number (starting from 1) that was marked as done. 
+     */
+    private static void displayMarkTaskSuccessMessage(int taskNumber) {
+        displayToUser(MESSAGE_MARKED, tasks[taskNumber].toString());
+    }
+
+    /**
+     * Adds a new todo task to task list. 
+     */
+    private static void executeAddTodo() {
+        tasks[sizeOfTaskList] = new Todo(userCommand);
+        sizeOfTaskList++;
+        displayAddTaskSuccessMessage();
+    }
+
+    /**
+     * Adds a new deadline task to task list. 
+     * Processes the user input to extract date. 
+     * If date is not null, create new deadline task. 
+     * Otherwise, display invalid command error. 
+     */
+    private static void executeAddDeadline() {
+        String date = processParameters(DEADLINE_DATA_PREFIX_BY);
+        if (date != null) {
+            tasks[sizeOfTaskList] = new Deadline(userCommand, date);
+            sizeOfTaskList++;
+            displayAddTaskSuccessMessage();
+        } else {
+            displayToUser(ERROR_INVALID_COMMAND_RECEIVED);
+        }
+    }
+
+    /**
+     * Adds a new event task to task list. 
```

Comment:
Would be good to keep the length of the comments consistent (e.g. executeAddDeadline and executeAddEvent) since they do similar actions.

Reply:
Oh thanks for noticing, Ill change it so it will look similar